From 927dbf93a67a347e4c8cd7c44d874885c92723cd Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Tue, 26 Aug 2014 13:47:55 +0800
Subject: [PATCH] intel_mid_dma: disable CHT's LPDMA support

Delete Cherrytrail's LPDMA ACPI ID and PCI ID in intel_mid_dma.
Make CHT's LPIO DMA enumerated with upstream dw driver.

Change-Id: I84c2c63d875683e1cdd46d97a755dbb8198caa73
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-2493
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
(cherry picked from commit 69abdfaef5b2ee24ae02012bfb63dfb685d47355)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/dma/intel_mid_dma.c |   39 +--------------------------------------
 1 file changed, 1 insertion(+), 38 deletions(-)

diff --git a/drivers/dma/intel_mid_dma.c b/drivers/dma/intel_mid_dma.c
index f268be8..3b917f8 100644
--- a/drivers/dma/intel_mid_dma.c
+++ b/drivers/dma/intel_mid_dma.c
@@ -47,8 +47,6 @@
 #define INTEL_BYT_LPIO2_DMAC_ID		0x0F40
 #define INTEL_BYT_DMAC0_ID		0x0F28
 #define INTEL_CHT_DMAC0_ID             0x22A8
-#define INTEL_CHT_LPIO1_DMAC_ID		0x2286
-#define INTEL_CHT_LPIO2_DMAC_ID		0x22C0
 
 #define LNW_PERIPHRAL_MASK_SIZE		0x20
 #define ENABLE_PARTITION_UPDATE		(BIT(26))
@@ -91,9 +89,7 @@ static int get_ch_index(int status, unsigned int base)
 static inline bool is_byt_lpio_dmac(struct middma_device *mid)
 {
 	return mid->pci_id == INTEL_BYT_LPIO1_DMAC_ID ||
-		mid->pci_id == INTEL_BYT_LPIO2_DMAC_ID ||
-		mid->pci_id == INTEL_CHT_LPIO1_DMAC_ID ||
-		mid->pci_id == INTEL_CHT_LPIO2_DMAC_ID;
+		mid->pci_id == INTEL_BYT_LPIO2_DMAC_ID;
 }
 
 static void dump_dma_reg(struct dma_chan *chan)
@@ -2111,13 +2107,6 @@ static struct pci_device_id intel_mid_dma_ids[] = {
 		INFO(6, 0, 2047, 0, 0, 1, 0, INTEL_BYT_LPIO1_DMAC_ID, &v1_dma_ops)},
 	{ PCI_VDEVICE(INTEL, INTEL_BYT_LPIO2_DMAC_ID),
 		INFO(6, 0, 2047, 0, 0, 1, 0, INTEL_BYT_LPIO2_DMAC_ID, &v1_dma_ops)},
-	/* Cherryview Low Speed Peripheral DMA */
-	{ PCI_VDEVICE(INTEL, INTEL_CHT_LPIO1_DMAC_ID),
-		INFO(6, 0, 2047, 0, 0, 1, 0, INTEL_CHT_LPIO1_DMAC_ID,
-			&v1_dma_ops)},
-	{ PCI_VDEVICE(INTEL, INTEL_CHT_LPIO2_DMAC_ID),
-		INFO(6, 0, 2047, 0, 0, 1, 0, INTEL_CHT_LPIO2_DMAC_ID,
-			&v1_dma_ops)},
 
 	{ 0, }
 };
@@ -2160,30 +2149,6 @@ struct intel_mid_dma_probe_info dma_cht_info = {
 	.pdma_ops = &v2_dma_ops,
 };
 
-struct intel_mid_dma_probe_info dma_cht1_info = {
-	.max_chan = 6,
-	.ch_base = 0,
-	.block_size = 2047,
-	.pimr_mask = 0,
-	.pimr_base = 0,
-	.dword_trf = 1,
-	.pimr_offset = 0,
-	.pci_id = INTEL_CHT_LPIO1_DMAC_ID,
-	.pdma_ops = &v1_dma_ops,
-};
-
-struct intel_mid_dma_probe_info dma_cht2_info = {
-	.max_chan = 6,
-	.ch_base = 0,
-	.block_size = 2047,
-	.pimr_mask = 0,
-	.pimr_base = 0,
-	.dword_trf = 1,
-	.pimr_offset = 0,
-	.pci_id = INTEL_CHT_LPIO2_DMAC_ID,
-	.pdma_ops = &v1_dma_ops,
-};
-
 static const struct dev_pm_ops intel_mid_dma_pm = {
 	.suspend_late = dma_suspend,
 	.resume_early = dma_resume,
@@ -2220,8 +2185,6 @@ static const struct acpi_device_id dma_acpi_ids[] = {
 	{ "DMA0F28", (kernel_ulong_t)&dma_byt_info },
 	{ "ADMA0F28", (kernel_ulong_t)&dma_byt_info },
 	{ "INTL9C60", (kernel_ulong_t)&dma_byt1_info },
-	{ "80862286", (kernel_ulong_t)&dma_cht1_info },
-	{ "808622C0", (kernel_ulong_t)&dma_cht2_info },
 	{ "ADMA22A8", (kernel_ulong_t)&dma_cht_info },
 	{ },
 };
-- 
1.7.9.5

